internal/db/datatypes.go

```diff
@@ -0,0 +1,160 @@
+package db
+
+import (
+	"fmt"
+	"github.com/google/uuid"
+	"net/url"
+	"regexp"
+)
+
+var (
+	ErrValue = fmt.Errorf("%w: invalid value for type", ErrData)
+)
+
+type Datatype struct {
+	ID            uuid.UUID
+	Name          string
+	Validator     Code
+	StorageFormat StorageFormat
+}
+
+type StorageFormat int64
+
+const (
+	BoolFormat StorageFormat = iota
+	IntFormat
+	StringFormat
+	FloatFormat
+	UUIDFormat
+)
+
+var storageFormat map[StorageFormat]interface{} = map[StorageFormat]interface{}{
+	BoolFormat:   false,
+	IntFormat:    int64(0),
+	StringFormat: "",
+	FloatFormat:  0.0,
+	UUIDFormat:   uuid.UUID{},
+}
+
+var nativeDatatypes map[uuid.UUID]Datatype = map[uuid.UUID]Datatype{
```

Comment:
what does "native" mean in this context?

Reply:
Was thinking that it meant the ones that come by default in go. Don't necessarily love this word.